src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdkAppEngineDevServer1.java

```diff
@@ -76,26 +72,46 @@ public void run(RunConfiguration config) throws AppEngineException {
     }
 
     List<String> jvmArguments = new ArrayList<>();
-    if (appengineWeb.isJava8()) {
-      jvmArguments.add("-Duse_jetty9_runtime=true");
-      jvmArguments.add("-D--enable_all_permissions=true");
-    } else {
-      // Add in the appengine agent
-      String appengineAgentJar = new File(
-              sdk.getJavaAppEngineSdkPath().toFile(), "agent/appengine-agent.jar")
-              .getAbsolutePath();
-      jvmArguments.add("-javaagent:" + appengineAgentJar);
-    } 
     arguments.addAll(DevAppServerArgs.get("address", config.getHost()));
     arguments.addAll(DevAppServerArgs.get("port", config.getPort()));
     if (config.getJvmFlags() != null) {
       jvmArguments.addAll(config.getJvmFlags());
     }
+    arguments.addAll(DevAppServerArgs.get("default_gcs_bucket", config.getDefaultGcsBucketName()));
+    // TODO: It's not really clear what the behavior here is when this is specified
+    arguments.addAll(DevAppServerArgs.get("runtime", config.getRuntime()));
+
+    // Arguments ignore by dev appserver 1
+    checkAndWarnIgnored(config.getAppYamls(), "appYamls");
+    checkAndWarnIgnored(config.getAdminHost(), "adminHost");
+    checkAndWarnIgnored(config.getAdminPort(), "adminPort");
+    checkAndWarnIgnored(config.getAuthDomain(), "authDomain");
+    checkAndWarnIgnored(config.getStoragePath(), "storagePath");
+    checkAndWarnIgnored(config.getLogLevel(), "logLevel");
+    checkAndWarnIgnored(config.getMaxModuleInstances(), "maxModuleInstances");
+    checkAndWarnIgnored(config.getUseMtimeFileWatcher(), "useMtimeFileWatcher");
+    checkAndWarnIgnored(config.getThreadsafeOverride(), "threadsafeOverride");
+    checkAndWarnIgnored(config.getPythonStartupScript(), "pythonStartupScript");
+    checkAndWarnIgnored(config.getPythonStartupArgs(), "pythonStartupArgs");
+    checkAndWarnIgnored(config.getCustomEntrypoint(), "customEntrypoint");
+    checkAndWarnIgnored(config.getAllowSkippedFiles(), "allowSkippedFiles");
+    checkAndWarnIgnored(config.getApiPort(), "apiPort");
+    checkAndWarnIgnored(config.getAutomaticRestart(), "automaticRestart");
+    checkAndWarnIgnored(config.getClearDatastore(), "clearDatastore");
+    checkAndWarnIgnored(config.getDevAppserverLogLevel(), "devAppserverLogLevel");
+    checkAndWarnIgnored(config.getSkipSdkUpdateCheck(), "skipSdkUpdateCheck");
 
     arguments.add("--allow_remote_shutdown");
     arguments.add("--disable_update_check");
-    if (appengineWeb.isJava8()) {
+    if (determineJavaRuntimeVersion(config.getServices()).equals("java8")) {
```

Comment:
since this is the only place that `determineJavaRuntimeVersion` is called would it make more sense to push all the logic into the method and have it just return a boolean? e.g. `bool requiresJava8(services)`